test/test_mps.py

```diff
@@ -8876,64 +8876,93 @@ def test_cpu_indices(self, device="mps"):
 
 class TestRNNMPS(TestCaseMPS):
     def test_lstm_1(self, device="mps", dtype=torch.float32):
+        for layers in [1] if product_version < 13.0 else [1, 2, 5]:
+            torch.random.manual_seed(42)
+            rnn = nn.LSTM(7, 4, layers, device="cpu")
+            input = torch.randn(2, 3, 7, device="cpu")
+            hx = torch.randn(layers, 3, 4, device="cpu")
+            cx = torch.randn(layers, 3, 4, device="cpu")
+
+            cpu_output, (cpu_hn, cpu_cn) = rnn(input, (hx, cx))
+
+            rnn = rnn.to(device)
+            input = input.to(device)
+            hx = hx.to(device)
+            cx = cx.to(device)
+            output, (hn, cn) = rnn(input, (hx, cx))
+
+            self.assertEqual(cpu_output, output)
+            self.assertEqual(cpu_hn, hn)
+            self.assertEqual(cpu_cn, cn)
+
+            # test batch_first
+            rnn = nn.LSTM(7, 4, layers, device="cpu", batch_first=True)
+            input = torch.randn(3, 2, 7, device="cpu")
+            hx = torch.randn(layers, 3, 4, device="cpu")
+            cx = torch.randn(layers, 3, 4, device="cpu")
+            cpu_output, (cpu_hn, cpu_cn) = rnn(input, (hx, cx))
+
+            rnn = rnn.to(device)
+            input = input.to(device)
+            hx = hx.to(device)
+            cx = cx.to(device)
+            output, (hn, cn) = rnn(input, (hx, cx))
+
+            self.assertEqual(cpu_output, output)
+            self.assertEqual(cpu_hn, hn)
+            self.assertEqual(cpu_cn, cn)
+
+    def test_lstm_backward(self, device="mps", dtype=torch.float32):
+        for layers in [1] if product_version < 13.0 else [1, 2, 5]:
+            lstm = nn.LSTM(2, 4, layers)  # initialized globally for consistent parameters init
+            lstm.train()
+
+            def get_results(device, inp, hx, cx):
+                rnn = lstm.to(device)
+                inp, hx, cx = inp.to(device), hx.to(device), cx.to(device)
+
+                output, _ = rnn(inp, (hx, cx))
+                f = output.sum()
+
+                weight_grads = sorted(
+                    [(param[0], torch.autograd.grad(f, [param[1]], retain_graph=True)[0]) for param in rnn.named_parameters()],
+                    key=lambda x: x[0]
```

Comment:
Could we do this with a single grad call instead?

Something like:
```
param_names, params = zip(*rnn.named_parameters())
param_grads = zip(param_names, torch.autograd.grad(out, params))
```

Otherwise, looks good. Thanks for updating!

Reply:
Sure, will fix this right now. Python's not my *mother tongue* ;)

Reply:
Still will leave

```python
input_grad, hx_grad, cx_grad = torch.autograd.grad(f, [inp, hx, cx])
```

It's possible to add these gradients to all other parameters, but I think it's better to keep `input_grad, hx_grad, cx_grad` apart from params for clarity

Reply:
Thanks @AlexRoar.